packages/go_router/example/lib/stateful_nested_navigation.dart

```diff
@@ -0,0 +1,420 @@
+                  path: 'details',
+                  builder: (BuildContext context, GoRouterState state) =>
+                      const DetailsScreen(label: 'A'),
+                ),
+              ],
+            ),
+          ),
+
+          /// The route branch for the second tab of the bottom navigation bar.
+          ShellRouteBranch(
+            /// It's not necessary to provide a navigatorKey if it isn't also
+            /// needed elsewhere. If not provided, a default key will be used.
+            // navigatorKey: _tabBNavigatorKey,
+            rootRoute: GoRoute(
+              /// The screen to display as the root in the second tab of the bottom
+              /// navigation bar.
+              path: '/b',
+              builder: (BuildContext context, GoRouterState state) =>
+                  const RootScreen(
+                      label: 'B',
+                      detailsPath: '/b/details/1',
+                      secondDetailsPath: '/b/details/2'),
+              routes: <RouteBase>[
+                GoRoute(
+                  path: 'details/:param',
+                  builder: (BuildContext context, GoRouterState state) =>
+                      DetailsScreen(label: 'B', param: state.params['param']),
+                ),
+              ],
+            ),
+          ),
+
+          /// The route branch for the third tab of the bottom navigation bar.
+          ShellRouteBranch(
+            /// Since this route branch has a nested StatefulShellRoute as the
+            /// root route, we need to specify what the default location for the
+            /// branch is.
+            defaultLocation: '/c1',
+            rootRoute: StatefulShellRoute.rootRoutes(
+              /// This bottom tab uses a nested shell, wrapping sub routes in a
+              /// top TabBar. In this case, we're using the `rootRoutes`
+              /// convenience constructor.
+              routes: <GoRoute>[
+                GoRoute(
+                  path: '/c1',
+                  builder: (BuildContext context, GoRouterState state) =>
+                      const TabScreen(label: 'C1', detailsPath: '/c1/details'),
+                  routes: <RouteBase>[
+                    GoRoute(
+                      path: 'details',
+                      builder: (BuildContext context, GoRouterState state) =>
+                          const DetailsScreen(label: 'C1', withScaffold: false),
+                    ),
+                  ],
+                ),
+                GoRoute(
+                  path: '/c2',
+                  builder: (BuildContext context, GoRouterState state) =>
+                      const TabScreen(label: 'C2', detailsPath: '/c2/details'),
+                  routes: <RouteBase>[
+                    GoRoute(
+                      path: 'details',
+                      builder: (BuildContext context, GoRouterState state) =>
+                          const DetailsScreen(label: 'C2', withScaffold: false),
```

Comment:
Without a scaffold, the transition animation won't work correctly:

![transition 2022-10-27 11_37_04](https://user-images.githubusercontent.com/1145719/198371519-77faf1a1-3a85-4d2b-bdd4-b839d5020df9.gif)

Reply:
Yeah, I believe this glitch is due to the lack of a background color when no Scaffold is used. I think a container for instance would also get the job done. A Scaffold is possibly simpler though, but I avoided that mostly due to the recommendation of not nesting Scaffolds.

Reply:
Where do you see that recommendation?

Reply:
Well, the [documentation for Scaffold](https://api.flutter.dev/flutter/material/Scaffold-class.html) states this: 

> Nested Scaffolds
> The **Scaffold is designed to be a top level container** for a MaterialApp. This means that adding a Scaffold to each route on a Material app will provide the app with Material's basic visual layout structure.
> 
> It is **typically not necessary to nest Scaffolds**.... 
> ...
> Although there are some use cases, like a presentation app that shows embedded flutter content, where nested scaffolds are appropriate, **it's best to avoid nesting scaffolds**.

From this, I get the impression that nesting scaffolds isn't recommended. But I guess the wording allows for exceptions, and that nested Scaffolds can be ok under some circumstances. And perhaps this is such a case? And if so, I see no problems using a Scaffold here.